### ISSUE TITLE
SAF provider does not show in pickers

### ISSUE THREAD

**Issue by chrysn** [termux/termux-app#1424]
**Problem description**

The SAF (Storage Access Framework) provider of termux only shows in the built-in file manager, but not in the file picker as used by applications like MaterialFiles.

**Steps to reproduce**

Install [MaterialFiles](https://f-droid.org/packages/me.zhanghai.android.files/) in parallel to termux, and make sure the native file manager shows "Termux" as a possible source of files in parallel to internal and external storage.

In MaterialFiles, when selecting "Add external storage" from the burger menu, all the other SAF providers show (including, if installed, NextCloud), but Termux is absent.

**Expected behavior**

Termux SAF would show on all SAF pickings.

**Additional information**

I had originally [reported this against MaterialFiles](https://github.com/zhanghai/MaterialFiles/issues/223), but they claim it's termux' fault -- easily possible, I'm not sufficiently well versed in Android details to tell.

Post output of command `termux-info`.

```
System information:
Linux localhost 3.10.84-g0befc21 #1 SMP PREEMPT Thu Feb 7 14:47:14 UTC 2019 aarch64 Android
Termux-packages arch:
aarch64
Android version:
7.1.2
Device manufacturer:
Sony
Device model:
E5823
```

**Comment by pazos** [comment]
> The SAF (Storage Access Framework) provider of termux only shows in the built-in file manager, but not in the file picker as used by applications like MaterialFiles.

Partially true. The content provider of Termux reacts to [ACTION_OPEN_DOCUMENT](https://developer.android.com/reference/android/content/Intent.html#ACTION_OPEN_DOCUMENT), which is part of SAF.

Most file managers use ACTION_OPEN_DOCUMENT_TREE (available since Lollipop) to grant RW permission on particular folders. And this is not supported in Termux.

I dunno if this (the absence of treeUri handling) was left out for some reason, but implementing it on top of current DocumentProvider should be possible.

**Comment by x0b** [comment]
I'd be interested in implementing this together with #1350.

**Comment by fornwall** [comment]
Fixed by @x0b (great work!) in #1460, which has just been released in version 0.92 of the app.
